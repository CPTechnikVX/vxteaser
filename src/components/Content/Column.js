import React     from 'react';
import Constants from '../../utils/Constants';
import PropTypes from 'prop-types';

/**
 * Column element for the grid
 */
export default class Column extends React.PureComponent {
	static propTypes = {
		/** @ignore */
		children: PropTypes.node,
		width:    PropTypes.number,

	};

	render() {
		const {children, width} = this.props;
		const styleObj          = {
			width: width ? width + '%' : 'auto',
			float: 'left',
		};

		return <div className={Constants.ClassName.Column} style={styleObj}>{children}</div>;
	}
}
