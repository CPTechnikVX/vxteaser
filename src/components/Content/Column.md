```html
<Col width="20%">This is a column of 20% width</Col>
```

```html
<Col>This is a column with an automatic width</Col>
```
