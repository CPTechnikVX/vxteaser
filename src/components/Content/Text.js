import React      from 'react';
import Constants  from '../../utils/Constants';
import classnames from 'classnames';
import PropTypes  from 'prop-types';

/**
 * Default free text element
 */
export default class Text extends React.PureComponent {
	static propTypes = {
		/** @ignore */
		children: PropTypes.node,
		color:    PropTypes.string,
	};

	render() {
		const {children, color} = this.props;
		const classList         = [];

		classList.push(Constants.ClassName.Text);

		if (color) {
			classList.push(Constants.PrefixClassName.Color + color);
		}

		return <span className={classnames(classList)}>{children}</span>;
	}
}
