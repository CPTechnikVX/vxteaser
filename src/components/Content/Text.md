```js
<Text color="highlight" modifier="-mb-20">Text with margin 20px bottom</Text>
```
