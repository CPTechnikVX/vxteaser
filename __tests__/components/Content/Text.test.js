import React     from 'react';
import {shallow} from 'enzyme';
import Column    from '../../../src/components/Content/Column';
import Text      from '../../../src/components/Content/Text';

describe('render text', () => {
	test('with content', () => {
		const wrapper = shallow(
			<Text>This is the text!</Text>
		);
		expect(wrapper).toMatchSnapshot();
	});
	test('with color', () => {
		const wrapper = shallow(
			<Text color="highlight">This is the text with highlight color!</Text>
		);
		expect(wrapper).toMatchSnapshot();
	});
});
