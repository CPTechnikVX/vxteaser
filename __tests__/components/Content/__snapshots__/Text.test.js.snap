// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render text with color 1`] = `
<span
  className="vxteaser-text vxteaser-color--highlight"
>
  This is the text with highlight color!
</span>
`;

exports[`render text with content 1`] = `
<span
  className="vxteaser-text"
>
  This is the text!
</span>
`;
